src/niweb/apps/noclook/management/commands/csvimport.py

```diff
@@ -0,0 +1,286 @@
+
+        csv_organizations = None
+        csv_contacts = None
+        self.user = User.objects.all().first()
+
+        # IMPORT ORGANIZATIONS
+        if options['organizations']:
+            # py: count lines
+            csv_organizations = options['organizations']
+            org_lines = self.count_lines(csv_organizations)
+
+            if options['verbosity'] > 0:
+                self.stdout.write('Importing {} Organizations from file "{}"'\
+                    .format(org_lines, csv_organizations.name))
+
+            total_lines = total_lines + org_lines
+
+        # IMPORT CONTACTS AND ROLES
+        if options['contacts']:
+            # py: count lines
+            csv_contacts = options['contacts']
+            con_lines = self.count_lines(csv_contacts)
+
+            if options['verbosity'] > 0:
+                self.stdout.write('Importing {} Contacts from file "{}"'\
+                    .format(con_lines, csv_contacts.name))
+
+            total_lines = total_lines + con_lines
+
+        imported_lines = 0
+        # print progress bar
+        if options['verbosity'] > 0:
+            self.printProgressBar(imported_lines, total_lines)
+
+        # process organizations
+        if options['organizations']:
+            # contact
+            node_type = NodeType.objects.filter(type=self.new_types[0]).first()
+            csv_organizations = options['organizations']
+            node_list = self.read_csv(csv_organizations, delim=self.delimiter)
+
+            for node in node_list:
+                account_name = node['account_name']
+
+                # dj: organization exist?: create or get (using just the name)
+                new_organization = self.get_or_create(
+                        account_name,
+                        node_type,
+                        relation_meta_type
+                    )
+
+            	# n4: add attributes
+                graph_node = new_organization.get_node()
+
+                graph_node.add_property('name', account_name)
+                for key in node.keys():
+                    if key not in ['account_name', 'parent_account'] and node[key]:
+                        graph_node.add_property(key, node[key])
+
+                	# dj: if parent organization: create or get (using just the name)
+                    if key == 'parent_account' and node['parent_account']:
+                        parent_org_name = node['parent_account']
+
+                        parent_organization = self.get_or_create(
+                                parent_org_name,
+                                node_type,
+                                relation_meta_type
+                            )
+
+                        parent_node = parent_organization.get_node()
+                        parent_node.add_property('name', parent_org_name)
+
+            	        # n4: add relation between org and parent_org
+                        graph_node.set_parent(parent_organization.pk)
```

Comment:
Not incorrect but I would explicitly use handle_id even though it is the NodeHandles primary key.